sdk/spring/scripts/ci_utils.py

```diff
@@ -0,0 +1,102 @@
+# some classes, methods support for ci_update_versions.py
+
+from enum import Enum
+import re
+
+include_update_marker = re.compile(r'\{x-include-update;([^;]+);([^}]+)\}')
+version_update_start_marker = re.compile(r'\{x-version-update-start;([^;]+);([^}]+)\}')
+version_update_end_marker = re.compile(r'\{x-version-update-end\}')
+version_update_marker = re.compile(r'\{x-version-update;([^;]+);([^}]+)\}')
+
+# regex for the version string is suggested one directly from semver.org
+# it's worth noting that both regular expressions on that page have start
+# of line (^) and end of line ($) anchors which need to be removed since
+# what's being matched is in the middle of the string
+# https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+version_regex_str_no_anchor = r'(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?'
+
+# External dependency versions do not have to match semver format and the semver regular expressions
+# will partially match and produce some hilarious results.
+external_dependency_include_regex = r'(?<=<include>).+?(?=</include>)'
+
+# External dependency versions do not have to match semver format and the semver regular expressions
+# will partially match and produce some hilarious results.
+external_dependency_version_regex = r'(?<=<version>).+?(?=</version>)'
+
+
+# Allow list prefix remover
+allowlist_exception_identifier_remover_regex = re.compile(r'^(?:.+_)(?=.+:)(.*)$')
+
+class UpdateType(Enum):
+    external_dependency = 'external_dependency'
+    library = 'library'
+    all = 'all'
+
+    # defining string is necessary to get ArgumentParser's help output to produce
+    # human readable values of UpdateType
+    def __str__(self):
+        return self.value
+
+    # defining string is necessary to get ArgumentParser's help output to produce
+    # human readable values of BuildType
+    def __str__(self):
+        return self.value
+
+class CodeModule:
+    def __init__(self, module_str):
+        # For library versions there will be up to 3 items resulting from the split
```

Comment:
We do not need this, because we only need to update external dependencies' version.